### PR TITLE
fix cache

### PR DESCRIPTION
**Pull request by medAmineSwi**

### CODE REVIEW

File: StylableSwiftUI/Classes/StylistIdentifier.swift [hunk 1]
```diff
@@ -186,3 +186,13 @@ private extension RandomAccessCollection  {
         self.count > 1 ? self[self.index(self.startIndex, offsetBy: 1)] : nil
     }
 }
+struct StyleCachKey: Hashable {
```
Comment (kerrmarin-lvmh) [inline StylableSwiftUI/Classes/StylistIdentifier.swift:189]:
```suggestion

struct StyleCacheKey: Hashable {
```

File: StylableSwiftUI/Classes/Stylist.swift [hunk 2]
```diff
@@ -39,7 +39,7 @@ public class Stylist: ObservableObject {
     }
 
     private var scoredStyleMatchCache: [StylistIdentifier: Style] = [:]
-    static var scoredImageStyleMatchCache: [StylistIdentifier: String] = [:]
+    static var scoredImageStyleMatchCache: [StyleCachKey: String] = [:]
```
Comment (kerrmarin-lvmh) [inline StylableSwiftUI/Classes/Stylist.swift:42]:
```suggestion
    static var scoredImageStyleMatchCache: [StyleCacheKey: String] = [:]
```

File: StylableSwiftUI/Classes/StylableImage.swift [hunk 3]
```diff
@@ -204,15 +204,15 @@ extension StylistIdentifier {
                  bundle: Bundle? = nil,
                  compatibleWith traits: UITraitCollection? = nil,
                  theme: Theme? = nil) -> UIImage? {
-        if let bestMatch = Stylist.scoredImageStyleMatchCache[self] {
+        if let bestMatch = Stylist.scoredImageStyleMatchCache[StyleCachKey(identifier: self, theme: theme)] {
```
Comment (kerrmarin-lvmh) [inline StylableSwiftUI/Classes/StylableImage.swift:207]:
```suggestion
        if let bestMatch = Stylist.scoredImageStyleMatchCache[StyleCacheKey(identifier: self, theme: theme)] {
```

File: StylableSwiftUI/Classes/StylistIdentifier.swift (hunk 1 above)
Comment (kerrmarin-lvmh) [reply StylableSwiftUI/Classes/StylistIdentifier.swift:189]: I would prefer this in its own file too

**Comment by medAmineSwi** [comment]
fixed

**Review by kerrmarin-lvmh — APPROVED** [review]

File: StyleCacheKey.swift [hunk 4]
```diff
@@ -0,0 +1,15 @@
+//
+//  StyleCacheKey.swift
+//
+
+import Foundation
+
+struct StyleCacheKey: Hashable {
+    var identifier:StylistIdentifier
+    var theme:Theme?
```
Comment (kerrmarin-lvmh) [inline StyleCacheKey.swift:9]:
```suggestion
    var identifier: StylistIdentifier
    var theme: Theme?
```

**Pull request merged**